videostore/utils.go

```diff
@@ -253,31 +257,31 @@ func sortFilesByDate(files []fileWithDate) {
 }
 
 // extractDateTimeFromFilename extracts the date and time from the filename.
+// Returns UTC time regardless of filename format:
+// - Unix timestamp filenames: directly converts to UTC time
+// - Datetime format filenames (YYYY-MM-DD_HH-mm-ss): parses as local time, converts to UTC
 func extractDateTimeFromFilename(filePath string) (time.Time, error) {
-	const minParts = 2
 	baseName := filepath.Base(filePath)
-	parts := strings.Split(baseName, "_")
-	if len(parts) < minParts {
-		return time.Time{}, fmt.Errorf("invalid file name: %s", baseName)
+	nameWithoutExt := strings.TrimSuffix(baseName, filepath.Ext(baseName))
+
+	// Unix timestamp case
+	if timestamp, err := strconv.ParseInt(nameWithoutExt, 10, 64); err == nil {
+		return time.Unix(timestamp, 0).UTC(), nil
 	}
-	datePart := parts[0]
-	timePart := strings.TrimSuffix(parts[1], filepath.Ext(parts[1]))
-	dateTimeStr := datePart + "_" + timePart
-	return ParseDateTimeString(dateTimeStr)
+
+	// Datetime format case
+	return ParseDateTimeString(nameWithoutExt)
 }
 
-// ParseDateTimeString parses a date and time string in the format "2006-01-02_15-04-05".
-// Returns a time.Time object and an error if the string is not in the correct format.
+// ParseDateTimeString parses a datetime string in our format "2006-01-02_15-04-05" from localtime,
+// converting it to UTC for internal processing.
 func ParseDateTimeString(datetime string) (time.Time, error) {
-	dateTime, err := time.Parse("2006-01-02_15-04-05", datetime)
+	//nolint:gosmopolitan // datetime format timestamps must be parsed into local time before UTC conversion.
+	t, err := time.ParseInLocation("2006-01-02_15-04-05", datetime, time.Local)
 	if err != nil {
-		return time.Time{}, err
+		return time.Time{}, fmt.Errorf("%w: %w", ErrInvalidDatetimeFormat, err)
```

Comment:
btw the NickSanfordGPT in my head says that we might want to assert on `time.ParseError` instead of our custom type, but I tried it out for a bit and got stuck on 

```
    utils_test.go:267: Expected error("parsing time "invalid" as "2006-01-02_15-04-05": cannot parse "invalid" as "2006"") to wrap error("parsing time "" as "": cannot parse "" as """) but it didn't.
```